frontend/javascripts/oxalis/model/sagas/min_cut_saga.js

```diff
@@ -0,0 +1,722 @@
+    }
+
+    if (!(distanceField[ll(neighborPos)] < currentDistance && distanceField[ll(neighborPos)] > 0)) {
+      throw new Error("Direction points towards a higher distance?");
+    }
+
+    const currDist = distanceField[ll(neighborPos)];
+    const distToSeed = Math.min(currDist, maxDistance - currDist);
+
+    // We don't want to remove voxels which are close to the seeds.
+    // See explanation for MIN_DIST_TO_SEED for details.
+    if (distToSeed > minDistToSeed) {
+      removedEdgeCount++;
+      path.unshift(neighborPos);
+
+      // Remove ingoing edge
+      removeIngoingEdge(edgeBuffer, ll(currentVoxel), neighborIdx);
+
+      // Remove outgoing edge
+      const invertedNeighborIdx = invertNeighborIdx(neighborIdx);
+      removeOutgoingEdge(edgeBuffer, ll(neighborPos), invertedNeighborIdx);
+    }
+
+    voxelStack.push(neighborPos);
+  }
+
+  return { path, foundSeed, removedEdgeCount };
+}
+
+function traverseResidualsField(
+  boundingBoxTarget: BoundingBox,
+  seedA: Vector3,
+  ll,
+  edgeBuffer: Uint16Array,
+) {
+  // Perform a breadth-first search from seedA and return
+  // which voxels were visited (visitedField). This represents
+  // the graph component for seedA.
+  const visitedField = new Uint16Array(boundingBoxTarget.getVolume());
+  const queue: Array<{ voxel: Vector3 }> = [{ voxel: seedA }];
+  while (queue.length > 0) {
+    const { voxel: currVoxel } = queue.shift();
+
+    const currVoxelIdx = ll(currVoxel);
+    if (visitedField[currVoxelIdx] > 0) {
+      continue;
+    }
+    visitedField[currVoxelIdx] = 1;
+    const neighbors = getNeighborsFromBitMask(edgeBuffer[currVoxelIdx]).outgoing;
+
+    for (const neighbor of neighbors) {
+      const neighborPos = V3.add(currVoxel, neighbor);
+
+      queue.push({ voxel: neighborPos });
+    }
+  }
+
+  return { visitedField };
+}
+
+function labelDeletedEdges(
+  visitedField,
+  boundingBoxTarget,
+  size,
+  originalEdgeBuffer,
+  targetMag,
+  l,
+  ll,
+) {
+  for (let x = 0; x < size[0]; x++) {
+    for (let y = 0; y < size[1]; y++) {
+      for (let z = 0; z < size[2]; z++) {
+        const idx = l(x, y, z);
+        if (visitedField[idx] === 1) {
+          const neighbors = getNeighborsFromBitMask(originalEdgeBuffer[idx]).outgoing;
+          const currentPos = [x, y, z];
+
+          for (const neighbor of neighbors) {
+            const neighborPos = V3.add(currentPos, neighbor);
+
+            if (visitedField[ll(neighborPos)] === 0) {
+              const position = V3.fromMagToMag1(
+                V3.add(boundingBoxTarget.min, neighborPos),
+                targetMag,
+              );
+              for (let dx = 0; dx < targetMag[0]; dx++) {
+                for (let dy = 0; dy < targetMag[1]; dy++) {
+                  for (let dz = 0; dz < targetMag[2]; dz++) {
+                    api.data.labelVoxels([V3.add(position, [dx, dy, dz])], 0);
+                  }
+                }
+              }
```

Comment:
Wouldn't it be equivalent to label the single voxel in the target mag and let the upsampling take care of the rest. Or is it needed that the voxels in mag 1 are immediately labeled?